vminitd/Sources/vmexec/RunCommand.swift

```diff
@@ -104,23 +142,27 @@ struct RunCommand: ParsableCommand {
                 }
             }
 
+            var fdCopy = hostFd
+            var fdData = Data(bytes: &fdCopy, count: MemoryLayout.size(ofValue: fdCopy))
+            try childPipe.fileHandleForWriting.write(contentsOf: fdData)
+            try childPipe.fileHandleForWriting.close()
+
+            var ackBuf: UInt8 = 0
+            _ = read(4, &ackBuf, 1)
+            _ = close(4)
```

Comment:
Once this is acked we should be able to close hostFd and we can remove the preserve logic